crates/analyzer/src/symbol.rs

```diff
@@ -297,6 +297,18 @@ impl Symbol {
             _ => Vec::new(),
         }
     }
+
+    pub fn proto(&self) -> Option<SymbolPath> {
+        match &self.kind {
+            SymbolKind::Module(x) => x.proto.clone(),
+            SymbolKind::Interface(_) => Some((&self.token).into()),
```

Comment:
Is this line necessary?

Reply:
Yes.
It is used at here.
https://github.com/veryl-lang/veryl/blob/c353e6138556dd4d5c13571e37eb9b3cbce7700d/crates/analyzer/src/handlers/check_type.rs#L274

Reply:
In this PR, interface can't be used as generic boudary. So I think adding ` SymbolKind::Interface(_)` is not necessary, and `SymbolKind::Interface(x) => x.proto.clone(),` should be added at introducing proto interface.

Reply:
The `inst` boundary with specific interface is needed for usecase shown in #793.
Therefore, this change is needed.

Reply:
I understood this line is used, but I think it is not appropriate that non proto symbol returns `Some` from `proto()`.
Changing `proto()` to `proto_or_inst()` or splitting it into `proto()` and `inst()` is better.
I prefer the splitting way because inst can be assigned to a generic parameter bound as proto if proto and inst are not distincted.

Reply:
I removed the statement for the `SymbolKind::Interface` but did not add the `inst` function.
Instead of the function, I added `resolve_inst_generic_arg_type` and `resolve_proto_generic_arg_type` functions to simplify resolving type of generic arguments.

Reply:
Thanks!
I think this is better than my idea.